src/compiler/scala/tools/nsc/typechecker/PatternTypers.scala

```diff
@@ -97,9 +97,9 @@ trait PatternTypers {
       // Dueling test cases: pos/overloaded-unapply.scala, run/case-class-23.scala, pos/t5022.scala
       // A case class with 23+ params has no unapply method.
       // A case class constructor may be overloaded with unapply methods in the companion.
-      // A case class maybe have its own custom unapply (so non-synthetic) scala/bug#11252
-      // Unapply methods aren't `isCaseApplyOrUnapply` in Scala 3 tasty/run/src-2/tastytest/TestColour.scala
-      else if (canElide && caseClass.isCase && !member.isOverloaded && (member == NoSymbol || member.isSynthetic))
+      // A case class may have its own custom unapply (so non-synthetic) pos/t11252.scala
+      // A case class may have its own custom unapply unrelated to deconstructing pos/t12250.scala
+      else if (canElide && caseClass.isCase && (!member.isOverloaded && (member == NoSymbol || member.isSynthetic) || member.alternatives.exists(_.isSynthetic) && caseClass.tpe =:= pt))
```

Comment:
Talking about this diff in passing @adriaanm mentioned `isApplicable` in Infer, which is already `import infer._`-ed above, so I should see if I can use that over `=:=`.

Reply:
Turned out not to be necessary. `=:=` is right, and I wrote the super-type/sub-type variations to prove that it's right (or at least convinced myself it's right... confirmation bias...).

I also reviewed how user-defined unapplies (and applies) are determined (#5816) and it's using Type#matchesType.  But, again, `=:=` looks to be correct while `<:<` isn't, which was my original thought.